### DISCUSSION TITLE
Using Automatic to serve another application.

### DISCUSSION THREAD

**Discussion by AlonzoTG** [vladmandic/sdnext#3610] [Q&A]
I had been using Automatic to serve images to another application. (AI Roguelite). It had been working but I needed to reboot my rig to upgrade the power supply in anticipation of upgrading my machine from AMD 5000 series to (eventually) AMD 9000 series. My procedure for starting this is to first pull Github, then run the webui.bat It does throw some errors but this time it barely works, the web front end comes up but the backend seems sluggish or broken, and AI roguelite can't use the backend at all anymore. What's going on?

**Comment by vladmandic** [comment]
> It does throw some errors but this time it barely works, the web front end comes up but the backend seems sluggish or broken, and AI roguelite can't use the backend at all anymore. What's going on?

how can i know whats going on if NO information is provided?

**Reply by AlonzoTG** [reply]
I'm a monkey at a keyboard. The sum total of my 40 years of experience with computers leaves me with nothing but this:  "Thing don't work." There are literally 1 million files on my computer right now. It was working before I rebooted my computer. All available versions of the client application do not work anymore. And your app shows no sign of having received any contact from the client. The only thing that happened between working and not was essentially a reboot and a git pull.   (ok, there was also a power supply upgrade in anticipation of a motherboard swap and a CPU cooling system failure due to bad connection somewhere that could have caused Real Problems if I had let it go another minute..)

**Reply by AlonzoTG** [reply]
I just spent a few hours figuring out how to update git app (which has nothing whatsoever to do with github...) to see if that will clear some errors your script was throwing. I had to do a file-search to dig out the git 2.14 version that was on my system for some reason and update the path environment variable to point to the freshly installed git...

**Reply by AlonzoTG** [reply]
I seem to have cured the error messages to the console but the thing is not generating a test image from the normal web ui. the button changes to "generate starting" but nothing happens. 


2024-12-05 16:03:59,961 | sd | DEBUG | launch | Server: alive=True requests=1671 memory=2.83/63.93 status='idle' task='' timestamp=None id='task(km7iirz91gnp704)' job=0 jobs=0 total=1 step=0 steps=0 queued=1 uptime=609 elapsed=600.91 eta=None progress=0
2024-12-05 16:04:03,084 | sd | DEBUG | shared_state | Requested continue

Clicking stop does not reset the state to "generate", It will switch to "generate" briefly but returns to "generate starting".

**Comment by AlonzoTG** [comment]
I did a motherboard swap (need to migrate AM4 -> AM5, currently using cheap temporary CPU...)  I read some of the wiki and learned the command to make the log file slightly less useless. Apparently it's trying to use the CPU to do inference despite the --use-cuda parameter (Correct behavior would have been to die with a precise error message stating why cuda could not be used) and despite having TWO GPUs (plus a G-core or two on the CPU to play with...)..   I wrapped webui.bat in another bat that scrubs the logs on launch and adds the two command line options mentioned. I did read the logs and could not find any useful information other than the confirmation that "thing don't work"... [sdnext.log](https://github.com/user-attachments/files/18050942/sdnext.log)

**Reply by AlonzoTG** [reply]
I've been chasing down dependency packages, It looks like the last time I messed with Cuda was about 5 years ago, I cleared out the old packages and now GPU rendering is operating on GPU2 which is ideal because that's the one I bought just for AI/robot apocalypse related stuff.  The confounding thing is that it is still not serving to the other application. =|  I have no idea, maybe the other app is just that broken....
